plugins/utils/comment-box.nix

```diff
@@ -0,0 +1,153 @@
+  luaName = "comment-box";
+  defaultPackage = pkgs.vimPlugins.comment-box-nvim;
+  description = ''
+    Clarify and beautify your comments and plain text files using boxes and lines.
+  '';
+
+  maintainers = [ lib.maintainers.elythh ];
+
+  settingsOptions = {
+    comment_style =
+      defaultNullOpts.mkEnum
+        [
+          "line"
+          "block"
+          "auto"
+        ]
+        "line"
+        ''
+          Select the type of comments.
+        '';
+
+    doc_width = defaultNullOpts.mkInt 80 ''
+      Width of the document.
+    '';
+    box_width = defaultNullOpts.mkInt 60 ''
+      Width of the boxes.
+    '';
+
+    borders = {
+      top = defaultNullOpts.mkStr "─" ''
+        Symbol used to draw the top border of a box.
+      '';
+
+      bottom = defaultNullOpts.mkStr "─" ''
+        Symbol used to draw the bottom border of a box.
+      '';
+
+      left = defaultNullOpts.mkStr "│" ''
+        Symbol used to draw the left border of a box.
+      '';
+
+      right = defaultNullOpts.mkStr "│" ''
+        Symbol used to draw the right border of a box.
+      '';
+
+      top_left = defaultNullOpts.mkStr "╭" ''
+        Symbol used to draw the top left corner of a box.
+      '';
+
+      top_right = defaultNullOpts.mkStr "╮" ''
+        Symbol used to draw the top right corner of a box.
+      '';
+
+      bottom_left = defaultNullOpts.mkStr "╰" ''
+        Symbol used to draw the bottom left corner of a box.
+      '';
+
+      bottom_right = defaultNullOpts.mkStr "╯" ''
+        Symbol used to draw the bottom right corner of a box.
+      '';
+
+    };
+
+    line_width = defaultNullOpts.mkInt 70 ''
+      Width of the lines.
+    '';
+
+    lines = {
+      line = defaultNullOpts.mkStr "─" ''
+        Symbol used to draw a line.
+      '';
+
+      line_start = defaultNullOpts.mkStr "─" ''
+        Symbol used to draw the start of a line.
+      '';
+
+      line_end = defaultNullOpts.mkStr "─" ''
+        Symbol used to draw the end of a line.
+      '';
+
+      title_left = defaultNullOpts.mkStr "─" ''
+        Symbol used to draw the left border of the title.
+      '';
+
+      tttle_right = defaultNullOpts.mkStr "─" ''
+        Symbol used to draw the right border of the title.
+      '';
+
+      outer_blank_lines_above = defaultNullOpts.mkBool false ''
+        Insert a blank line above the box.
+      '';
+
+      outer_blank_lines_below = defaultNullOpts.mkBool false ''
+        Insert a blank line below the box.
+      '';
+
+      inner_blank_lines = defaultNullOpts.mkBool false ''
+        Insert a blank line above and below the text.
+      '';
+
+      line_blank_line_above = defaultNullOpts.mkBool false ''
+        Insert a blank line above the line.
+      '';
+
+      line_blank_line_below = defaultNullOpts.mkBool false ''
+        Insert a blank line below the line.
+      '';
+    };
+  };
+
+  settingsExample = {
+    comment_style = "line";
+    doc_width = 80; # width of the document
```

Comment:
We should probably remove all these comments, too. They won't be rendered in the page that gets created and they look like repeats of the descriptions for the options above that will.

Reply:
oh yes, I thought the raw nix code was shown in the example

Reply:
https://github.com/nix-community/nixvim/compare/3dac1a6e16d53c29d540cf2dc3405265cac1df66..c4c237e55eac9e5bdeb41edfcc004c2bca5b876a

Reply:
If you _want_ raw nix code to show up in the example, you can use `lib.literalExpression` (or `lib.literalMD` for injecting markdown).

e.g.

```nix
settingsExample = lib.literalExpression ''
  {
    comment_style = "line";
    doc_width = 80; # width of the document
    # etc
  }
'';

Most of the time we prefer just passing a nix value instead of a literalExpression string, since it ensures a somewhat standardized formatting, and ensures the syntax is valid, but I've nothing against using `literalExpression` where it makes sense to do so.